### ISSUE TITLE
DynamoDb "batch-write-item" fails with status code 500

### ISSUE THREAD

**Issue by chaz-doyle-cko** [localstack/localstack#3959]
This is a [x] bug report

# Detailed description
When trying to clear entries for integration tests we use the `batch-write-items` command on dynamodb. The `latest` version of localstack is failing to process the command and returning a `500 - Interval Server Error` after 9 retries.

## Expected behavior
Previous version (localstack/localstack:0.12.9.1) would successfully delete items from dynamodb.

## Actual behavior
Command either times out or returns a `500 - Internal Server Error`

# Steps to reproduce

```
docker run -d --name localstack-bug -e SERVICES=dynamodb localstack/localstack:latest
# wait for dynamodb to be ready

docker exec -ti localstack-bug sh

aws configure
# AWS Access Key ID [None]: xxx
# AWS Secret Access Key [None]: xxx
# Default region name [None]: eu-north-1
# Default output format [None]:

aws --endpoint-url http://localhost:4566 \
    dynamodb create-table \
    --table-name "example-table" \
    --attribute-definitions AttributeName=pk,AttributeType=S AttributeName=sk,AttributeType=S \
    --key-schema AttributeName=pk,KeyType=HASH AttributeName=sk,KeyType=RANGE \
    --provisioned-throughput ReadCapacityUnits=10,WriteCapacityUnits=5

aws --endpoint-url http://localhost:4566 \
     dynamodb put-item \
    --table-name example-table \
    --item '{"pk": {"S": "example"},"sk":{"S":"value"}}'

aws --endpoint-url http://localhost:4566 \
    dynamodb get-item \
    --table-name example-table \
    --key '{"pk": {"S": "example"},"sk":{"S":"value"}}'

aws --endpoint-url http://localhost:4566 \
    dynamodb batch-write-item \
    --request-items '{"example-table":[{"DeleteRequest":{"Key":{"pk": {"S": "example"},"sk":{"S":"value"}}}}]}'


# RESULT: An error occurred (500) when calling the BatchWriteItem operation (reached max retries: 9):
```

### Errors in localstack docker logs
```
2021-05-04T03:46:52:WARNING:localstack.utils.server.http2_server: Error in proxy handler for request POST http://localstack:4566/: local variable 'existing_item' referenced before assignment Traceback (most recent call last):
File "/opt/code/localstack/localstack/utils/server/http2_server.py", line 129, in index
raise result
File "/opt/code/localstack/localstack/utils/bootstrap.py", line 689, in run
result = self.func(self.params)
File "/opt/code/localstack/localstack/utils/async_utils.py", line 28, in _run
return fn(*args, **kwargs)
File "/opt/code/localstack/localstack/services/generic_proxy.py", line 352, in handler
response = modify_and_forward(method=method, path=path_with_params, data_bytes=data, headers=headers,
File "/opt/code/localstack/localstack/services/generic_proxy.py", line 171, in modify_and_forward
listener_result = listener.forward_request(method=method,
File "/opt/code/localstack/localstack/services/edge.py", line 118, in forward_request
return do_forward_request(api, method, path, data, headers, port=port)
File "/opt/code/localstack/localstack/services/edge.py", line 137, in do_forward_request
result = do_forward_request_inmem(api, method, path, data, headers, port=port)
File "/opt/code/localstack/localstack/services/edge.py", line 157, in do_forward_request_inmem
response = modify_and_forward(method=method, path=path, data_bytes=data, headers=headers,
File "/opt/code/localstack/localstack/services/generic_proxy.py", line 241, in modify_and_forward
updated_response = update_listener.return_response(**kwargs)
File "/opt/code/localstack/localstack/services/dynamodb/dynamodb_listener.py", line 337, in return_response
records, unprocessed_items = self.prepare_batch_write_item_records(record, data)
File "/opt/code/localstack/localstack/services/dynamodb/dynamodb_listener.py", line 524, in prepare_batch_write_item_records
new_record['dynamodb']['SizeBytes'] = len(json.dumps(existing_item))
UnboundLocalError: local variable 'existing_item' referenced before assignment
```

**Comment by sixteni** [comment]
If think the problem is here:
new_record['dynamodb']['SizeBytes'] = len(json.dumps(existing_item))
if think it should be:
new_record['dynamodb']['SizeBytes'] = len(json.dumps(existing_items[i]))

https://github.com/localstack/localstack/commit/0eb1df9f393a6d9c104ca55cc3f87fd6df743184#diff-7fd0be67bf91aadc4f382cd5a9aea9b818eca5639cd335723a659e2aee411b09

**Comment by lacerdaeduardo** [comment]
I'm running on the same problem. The `batch-write-item` always results on :

```
        if http.status_code >= 300:
            error_code = parsed_response.get("Error", {}).get("Code")
            error_class = self.exceptions.from_code(error_code)
>           raise error_class(parsed_response, operation_name)
E           botocore.exceptions.ClientError: An error occurred (500) when calling the BatchWriteItem operation (reached max retries: 9):
```

I'm using the latest version of localstack with python and boto3:
```python
all_items = dynamo_table.scan()
with dynamo_table.batch_writer() as batch:
        for item in all_items["Items"]:
            batch.delete_item(
                 Key={"user_id": item["user_id"], "message_id": item["message_id"]}
            )
```

**Comment by whummer** [comment]
Thanks @chaz-doyle-cko @sixteni @lacerdaeduardo @KrishnanRanjithkumar . This should be fixed in #3960 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists. Thanks!
